api/tests/mocha/services/africas-talking.spec.js

```diff
@@ -0,0 +1,161 @@
+const chai = require('chai');
+const sinon = require('sinon');
+const request = require('request-promise-native');
+const config = require('../../../src/config');
+const service = require('../../../src/services/africas-talking');
+const environment = require('../../../src/environment');
+const lib = { SMS: { send: () => {} } };
+
+describe('africas talking service', () => {
+
+  beforeEach(() => {
+    sinon.stub(service, '_getLib').returns(lib);
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  describe('send', () => {
+
+    it('fails early with bad config', done => {
+      sinon.stub(config, 'get').returns({});
+      const given = [ { uuid: 'a', to: '+123', content: 'hello' } ];
+      service.send(given)
+        .then(() => done(new Error('expected error to be thrown')))
+        .catch(err => {
+          chai.expect(err).to.equal('No username configured. Refer to the Africa\'s Talking configuration documentation.');
+          done();
+        });
+    });
+
+    it('forwards messages to lib', () => {
+      sinon.stub(environment, 'serverUrl').value('server.com');
+      sinon.stub(request, 'get').resolves('"555"\n');
+      sinon.stub(config, 'get').returns({
+        reply_to: '98765',
+        africas_talking: { username: 'user' }
+      });
+      sinon.stub(lib.SMS, 'send').resolves({
+        SMSMessageData: {
+          Message: 'Sent to 1/1 Total Cost: KES 0.8000',
+          Recipients: [{
+            statusCode: 101,
+            number: '+254711XXXYYY',
+            status: 'Success',
+            cost: 'KES 0.8000',
+            messageId: 'ATPid_SampleTxnId123'  
```

Comment:
Wow... what does the "P" stand for? Provider?

Reply:
Hah. No idea. Provisional?